.github/workflows/default.yaml

```diff
@@ -94,7 +94,12 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        os: [ ubuntu-22.04 ]
+        os: [ ubuntu-22.04, macos-latest ]
+        layer:
+          - layer-00-default
+          - layer-01-minimal
+          - layer-02-maximus
+          - layer-04-noauth-everything
```

Comment:
Please move these out-of-scope changes into a dedicated PR. Since these changes create a lot more CI noise and duplicate information, that dedicated PR should provide data to back its "speed up" claims.

Reply:
https://github.com/kinkie/squid/actions has a lot of output from experiments.

[MacOS serial run with a cold ccache](https://github.com/kinkie/squid/actions/runs/8735961320) - what is currently being done on Linux, takes 1h50
[MacOS serial run without ccache](https://github.com/kinkie/squid/actions/runs/8737350702) takes 1h35. Ccache would help but it is limited to a total 10gb per repository, not enough to hold all the variants we might want to support. With a hot ccache, build time is cut by about 75% ([single layer took 13 minutes](https://github.com/kinkie/squid/actions/runs/8725390026) [no-ccache is 1hr](https://github.com/kinkie/squid/actions/runs/8721931841) ).
[Parallel layer run, no ccache](https://github.com/kinkie/squid/actions/runs/8738417302) takes just short of 20 minutes, an 80% improvement in wall time against the 1h50 baseline

Reply:
Note, these times are subject to wide variations, as it depends on which workers the jobs happen to be scheduled and the load of these workers, but I hope I managed to convince you that they're beneficial.

CPU time increases are minimal, mostly related to the duplicate time spent installing dependencies

Reply:
Extracted in PR #1787

Reply:
Removed from here